src/server.rs

```diff
@@ -30,11 +35,11 @@ macro_rules! try_client {
 
 pub struct Hopper {
     metrics: Arc<Metrics>,
-    router: Arc<dyn Router>,
+    router: Arc<Mutex<Arc<dyn Router>>>,
```

Comment:
1. Adding a Mutex here would highly impact performance. I specifically chose to not use a Mutex here in the design of hopper
2. There's no need for a double Arc